torch/ao/sparsity/sparsifier/base_sparsifier.py

```diff
@@ -177,6 +178,14 @@ def convert(self):
         raise NotImplementedError('`convert` is not implemented. '
             'Please, use `torch.ao.utils.convert` instead.')
 
+    def step(self, use_path=True):
+        if not self.enable_mask_update:
+            return
+        with torch.no_grad():
+            for config in self.module_groups:
+                module = config['module']
+                self.update_mask(module, **config)
+
     @abc.abstractmethod
-    def step(self):
-        return
+    def update_mask(self, layer, **kwargs):
```

Comment:
Lets call this function as step instead of update mask. The reason is that:
 For every call of forward, we want the sparsifier to be called. This is to allow for updating of any stats used for mask calculation: This could be gradients, weight norms, activations etc.
The rate at  which the mask is updated can be controlled by the scheduler. At a high level, we do this:
``` python
def step(self):
  # Always update stats
  # Depending on scheduler, call self.update_mask
```